JSDemos/Demos/Toolbar/Multiline/React/App.js

```diff
@@ -0,0 +1,339 @@
+import React from 'react';
+import Toolbar, { Item } from 'devextreme-react/toolbar';
+import { Button } from 'devextreme-react/button';
+import { ButtonGroup } from 'devextreme-react/button-group';
+import RadioGroup from 'devextreme-react/radio-group';
+import Resizable from 'devextreme-react/resizable';
+import CheckBox from 'devextreme-react/check-box';
+import DateBox from 'devextreme-react/date-box';
+import DropDownButton from 'devextreme-react/drop-down-button';
+import SelectBox from 'devextreme-react/select-box';
+import notify from 'devextreme/ui/notify';
+import 'devextreme/ui/select_box';
+import {
+  fontSizes,
+  lineHeights,
+  fontFamilies,
+  fontStyles,
+  textAlignItems,
+  textAlignItemsExtended,
+  listTypes,
+} from './data.js';
+
+const lineHeightDefault = lineHeights[1].lineHeight;
+const textAlignDefault = [textAlignItems[0].alignment];
+const fontSizeDefault = fontSizes[2].size;
+
+const toolbarLineModes = [
+  {
+    text: 'Multiline mode',
+    value: true,
+  },
+  {
+    text: 'Singleline mode',
+    value: false,
+  },
+];
+
+function App() {
+  const onAttachButtonClick = React.useCallback(() => {
+    notify('Attach button has been clicked!');
+  }, []);
+
+  const onAddButtonClick = React.useCallback(() => {
+    notify('Add button has been clicked!');
+  }, []);
+
+  const onRemoveButtonClick = React.useCallback(() => {
+    notify('Remove button has been clicked!');
+  }, []);
+
+  const onAboutButtonClick = React.useCallback(() => {
+    notify('About button has been clicked!');
+  }, []);
+
+  const [attachButtonOptions] = React.useState({
```

Comment:
Why so we use state here?

Reply:
There is no need to use state for data that never changes. If you're never going to call setState you might as well just make it a simple const.
In cases where you need to save or pass external data you can use useRef instead.

Reply:
https://github.com/DevExpress/devextreme-demos/pull/2004/commits/b9a25ae2158bd511871b00651d19494c2ce4922e